app/controllers/application_controller.rb

```diff
@@ -1,3 +1,20 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
+
+  helper_method :current_user, :signed_in?
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+  
+  def signed_in?
+    !!current_user
+  end
+  
+  def require_user
+    if !signed_in?
+      flash[:error] = "You must be signed in."
```

Comment:
You'll have to change your `flash` class to match the current bootstrap 3 classes. Example `flash[:danger]`. See [here](http://getbootstrap.com/components/#alerts)